cli/internal/scope/filter/filter.go

```diff
@@ -39,16 +52,62 @@ func (r *Resolver) GetPackagesFromPatterns(patterns []string) (util.Set, error)
 		if err != nil {
 			return nil, err
 		}
-		selectors = append(selectors, &selector)
+		selectors = append(selectors, selector)
 	}
-	selected, err := r.GetFilteredPackages(selectors)
+	selected, err := r.getFilteredPackages(selectors)
 	if err != nil {
 		return nil, err
 	}
 	return selected.pkgs, nil
 }
 
-func (r *Resolver) GetFilteredPackages(selectors []*TargetSelector) (*SelectedPackages, error) {
+func (pi *PackageInference) apply(selector *TargetSelector) error {
+	if selector.namePattern != "" {
+		// The selector references a package name, don't apply inference
+		return nil
+	}
+	if pi.PackageName != "" {
+		selector.namePattern = pi.PackageName
+	}
+	if selector.parentDir != "" {
+		parentDir := pi.DirectoryRoot.UntypedJoin(selector.parentDir).ToStringDuringMigration()
```

Comment:
Good catch. We unfortunately have not plumbed through our paths here yet. 

Looking through the code, specifying an absolute path will break currently, or at least not match anything. We join it with the repo root / cwd regardless. But that does mean the above behavior is incorrect, since if we have a `parentDir`, it is always going to be an absolute path, and the correct behavior for inference is to join what the user specified to the inferred directory.

I'm going to take a stab at using our path types to sort this out and hope it doesn't spiral out too much.

Reply:
Ported `parentDir` on `TargetSelector` to `RelativeSystemPath`, and pulled the thread to doing the right thing here.

It wasn't possible for *the user input* to `parentDir` to be an absolute path before, but that was not obvious.